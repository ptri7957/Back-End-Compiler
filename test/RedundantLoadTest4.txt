( (branches (a)
    (0 (ld r1 a)
       (lc r2 0)
       (eq r3 r1 r2)
       (st c r3)
       (lc r4 0)
       (st b r4)
       (ld r5 c)
       (br r5 1 2)
       (ld r6 c)
       (ret r6) )
    (1 (ld r7 a)
       (lc r8 10)
       (add r9 r7 r8)
       (st b r9)
       (ld r10 b)
       (call r11 branches r10)
       (st b r11)
       (ld r12 b)
       (ret r12) )
    (2 (ld r13 a)
       (lc r14 100)
       (mul r15 r13 r14)
       (st b r15)
       (ld r16 b)
       (call r17 branches r16)
       (st b r17)
       (ld r18 b)
       (ret r18) ) ) )