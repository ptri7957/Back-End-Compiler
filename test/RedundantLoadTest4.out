( (branches (a)
    (0 (ld r1 a)
       (lc r2 0)
       (eq r3 r1 r2)
       (st c r3)
       (lc r4 0)
       (st b r4)
       (ld r5 c)
       (br r5 1 2) )
    (2 (lc r8 10)
       (add r9 r1 r8)
       (st b r9)
       (st b r11)
       (ld r12 b)
       (ret r12) )
    (3 (lc r14 100)
       (mul r15 r1 r14)
       (st b r15)
       (st b r17)
       (ld r18 b)
       (ret r18) ) ) )